src/components/EditableCounter.vue

```diff
@@ -1,135 +1,106 @@
+          v-bind:placeholder="this.counter.name"
+          v-model="counterName"
       />
     </label>
-    <button id="increase" type="button" class="btn btn-primary rounded-circle">
+    <button id="increase" type="button" class="btn btn-primary rounded-circle counter-button disabled">
       +
     </button>
     <label id="counter-max">
       <input
-        type="text"
-        v-bind:placeholder="this.counter.maxCount"
-        v-model="maxCount"
+          type="number"
+          v-bind:placeholder="this.counter.maxCount"
+          v-model="maxCount"
       />
     </label>
-    <button id="decrease" type="button" class="btn btn-primary rounded-circle">
+    <button id="decrease" type="button" class="btn btn-primary rounded-circle counter-button disabled">
       ─
     </button>
-    <button @click="displayAlert = true" id="delete-button">X</button>
+    <modal modal-name="confirm-modal">
+      <template v-slot:modal-body class="modal-body">
+        <h5>Are you sure you want to delete this counter?</h5>
+      </template>
+      <template v-slot:modal-footer>
+        <div class="footer">
+          <button class="btn btn-danger modal-button" @click="removeCounter" data-dismiss="modal">Yes!</button>
+          <button class="btn btn-dark modal-button" data-dismiss="modal">No</button>
+        </div>
+      </template>
+    </modal>
+    <button id="delete-button" class="rounded-1 btn btn-primary" data-toggle="modal" data-target="#confirm-modal">─
+    </button>
   </div>
+
 </template>
 
 <script lang="ts">
-import Vue, { VueConstructor } from "vue";
-import { CounterModel } from "@/types/CounterModel";
-import { mapState } from "vuex";
+import Vue, {VueConstructor} from "vue";
+import {CounterModel} from "@/types/CounterModel";
+import {mapState} from "vuex";
+import Modal from "@/components/Modal.vue";
 
 interface EditableData {
   counterName: string;
   maxCount?: number;
-  isValidNumber: boolean;
   displayAlert: boolean;
-  deleted: boolean;
 }
 
 export default (Vue as VueConstructor).extend({
   name: "EditableCounter",
-  props: { index: Number },
+  components: {Modal},
+  props: {index: Number},
   data(): EditableData {
     return {
       counterName: "",
       maxCount: undefined,
-      isValidNumber: true,
       displayAlert: false,
-      deleted: false,
     };
   },
   methods: {
-    removeCounter: function() {
-      this.deleted = true;
-    },
+    removeCounter(): void {
+      this.$store.commit("removeCounter", this.index)
+    }
   },
   computed: {
-    counter: function() {
+    counter: function () {
       return this.$store.getters.counterByIndex(this.index);
     },
     ...mapState(["mode"]),
   },
   watch: {
-    maxCountLocal(newValue: number) {
+    maxCount(newValue: string, oldValue?: string) {
       if (isNaN(Number(newValue))) {
-        this.isValidNumber = false;
-        this.displayAlert = true;
+        if (!oldValue || (newValue.length > oldValue.length))
```

Comment:
can you change the value of the input to 0 or something after displaying the message?